contentpal/src/test/java/org/dmfs/android/contentpal/utils/ContentOperationBuilderCountMatcher.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.contentpal.utils;
+
+import android.content.ContentProviderOperation;
+
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+import java.lang.reflect.Field;
+import java.util.Locale;
+
+
+/**
+ * A {@link ContentProviderOperation.Builder} {@link Matcher} which matches the operation type.
+ * <p>
+ * TODO: match Uri
+ *
+ * @author Marten Gajda
+ */
+public final class ContentOperationBuilderCountMatcher extends TypeSafeDiagnosingMatcher<ContentProviderOperation.Builder>
+{
+
+    private final Integer mExpectedCount;
+
+
+    public static ContentOperationBuilderCountMatcher expectingCount(int expectedCount)
+    {
+        return new ContentOperationBuilderCountMatcher(expectedCount);
+    }
+
+
+    public static ContentOperationBuilderCountMatcher expectingNoCount()
+    {
+        return new ContentOperationBuilderCountMatcher(null);
+    }
+
+
+    public ContentOperationBuilderCountMatcher(Integer expectedCount)
+    {
+        mExpectedCount = expectedCount;
+    }
+
+
+    @Override
+    protected boolean matchesSafely(ContentProviderOperation.Builder builder, Description mismatchDescription)
+    {
+        // create a builder an test the ContentValues
+        try
+        {
+            Field valuesField = ContentProviderOperation.Builder.class.getDeclaredField("mExpectedCount");
+            valuesField.setAccessible(true);
+
+            Integer expectedCount = (Integer) valuesField.get(builder);
+
+            if (mExpectedCount == null && expectedCount != null)
+            {
+                mismatchDescription.appendText(String.format(Locale.ENGLISH, "expects %d results", expectedCount));
```

Comment:
Same texts are repeated multiple times, could they be re-used?

Reply:
I guess so.